add product diags to 'working with support' (#1921)
elastic/docs-content: troubleshoot/index.md

  If you contact us by email, use the email address you registered with so we can help you more quickly. If your registered email is a distribution list, you can register a second email address with us. Just open a case to let us know the name and email address you want to add.
  :::

  :::{warning}

>   All cases opened by email are opened at normal severity. For incidents, kindly ensure to raise via the [Elastic Support Portal](https://support.elastic.co/) at the [appropriate severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case). 

Reviewer: What about this? Generally you want to avoid terms like "kindly" as a direct instruction is shorter and clearer.

```suggestion
  All cases opened by email default to normal severity. For incidents, open a case through the [Elastic Support Portal](https://support.elastic.co/) and select the [appropriate severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case). 
```

  :::

## Working with support [troubleshoot-work-with-support]

> When opening a support case, kindly be sure to:
> 
> * Describe the problem along with its [contextual severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case). 

Reviewer: I would just tighten this up a bit:

```suggestion
When you open a support case:

* Describe the problem and its [contextual severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case). 
```

  * If an error message was encountered, include full error message and timezone-designated dates and times of when the problem occurred.

  * If the problem is UI-related and does not replicate via API, pull a [browser network log](https://www.elastic.co/blog/generating-browser-har-file-kibana-troubleshooting). 

* Upload the related product's diagnostics and debug logs

>   * If {{serverless-full}} or {{ech}}, Support can pull diagnostics on your behalf as long as you include resource ids:

Reviewer: This renders as "If Elastic Cloud Serverless or Elastic Cloud Hosted.."

Reviewer: Ahh, I see below.

```suggestion
  * If hosting on {{serverless-full}} or {{ech}}, Support can pull diagnostics on your behalf as long as you include resource ids:
```

    
    * {{serverless-full}} Project ID or {{kib}} URL
    * {{ech}} Deployment ID or {{kib}} URL

>   * If hosting on {{ece}} or {{eck}}:

Reviewer: Does "Support can pull diagnostics on your behalf as long as you include resource ids:" apply here too? It's not clear to me if it does or doesn't.

Author: The resource ids apply below that ECE you have to "flagging `--deployments` as applicable" and ECK does pull everything it can see by default.

    * {{ece}} [diagnostic](/troubleshoot/deployments/cloud-enterprise/run-ece-diagnostics-tool.md) flagging `--deployments` as applicable
    * {{eck}} [diagnostic](/troubleshoot/deployments/cloud-on-k8s/run-eck-diagnostics.md)

  * If self-hosting:

    * {{es}} [diagnostic](troubleshoot/elasticsearch/diagnostic.md) and [debug logs](deploy-manage/monitor/logging-configuration/update-elasticsearch-logging-levels.md)
    * {{kib}} and {{fleet}} [diagnostic](troubleshoot/kibana/capturing-diagnostics.md) and [debug logs](/docs/deploy-manage/monitor/logging-configuration/kibana-log-levels.md)
    
    * Ingest tools

      * {{ls}} [diagnostic](troubleshoot/ingest/logstash/diagnostic.md) and [debug logs](reference/logstash/logging.md)
      * {{agent}} [diagnostic](https://www.elastic.co/docs/troubleshoot/ingest/fleet/common-problems#trb-collect-agent-diagnostics) and [debug logs](https://www.elastic.co/docs/reference/fleet/monitor-elastic-agent#change-logging-level)
  

  :::{tip}

>   {{es}}'s cluster maintains an advanced task management system, but the other products have a simpler polling mechanism. This can cause issues to only surface during their start-up debug logs as sequential logs will only note that the sub-process as stopped or that it has not changed state from earlier error.

Reviewer: In tech docs, product names are treated more like attributive nouns and shouldn't be possessive. So instead of `{{es}}'s cluster`, you should say something like `The {{es}} cluster`. 

What about this?

```suggestion
  The {{es}} cluster maintains an advanced task management system, while other products use a simpler polling mechanism. Because of this, some issues appear only in their start-up debug logs and later logs may only note that the subprocess has stopped or that it has not changed state from earlier error.
```

  If you contact us by email, use the email address you registered with so we can help you more quickly. If your registered email is a distribution list, you can register a second email address with us. Just open a case to let us know the name and email address you want to add.
  :::

  :::{warning}

>   All cases opened by email default to normal severity. For incidents, open a case through the [Elastic Support Portal](https://support.elastic.co/) and select the [appropriate severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case). 

Author: ```suggestion
  All cases opened by email default to a normal severity level. For incidents, open a case through the [Elastic Support Portal](https://support.elastic.co/) and select the [appropriate severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case).
```

>   All cases opened by email default to normal severity. For incidents, open a case through the [Elastic Support Portal](https://support.elastic.co/) and select the [appropriate severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case). 

Reviewer 2: Thanks!

  :::

## Working with support [troubleshoot-work-with-support]

When you open a support case:

> * Describe the problem and its [contextual severity](https://www.elastic.co/support/welcome#what-to-say-in-a-case). 

Reviewer 2: ```suggestion
* Describe the problem with context about the situation, severity, and seriousness, as described in [What should I say in my case?](https://www.elastic.co/support/welcome#what-to-say-in-a-case).
```
"Contextual severity" isn't super clear to me so how about the above, to better match the article we're linking to?

Reviewer 2: Ship it! ⛵

>   * If an error message was encountered, include full error message and timezone-designated dates and times of when the problem occurred.

Reviewer 2: ```suggestion
  * If an error message was encountered, include the full error message, a timezone-designated date, and the time of when the problem occurred.
```
I'd keep everything singular. But if there's typically more than one problem we could make all of these plural.

Author: ```suggestion
  * If an error message was encountered, include the full error message and the timezone-designated timestamp of when the problem occurred.
```
is this variation okay?

Reviewer 2: Looks great. Thanks!

  * If the problem is UI-related and can't be replicated using API calls, pull a [browser network log](https://www.elastic.co/blog/generating-browser-har-file-kibana-troubleshooting). 

* Upload the related product's diagnostics and debug logs:

>   * If hosting on {{serverless-full}} or {{ech}}, Support can pull diagnostics on your behalf as long as you include resource IDs:

Reviewer 2: ```suggestion
  * If you're using {{serverless-full}} or {{ech}}, Support can pull diagnostics on your behalf as long as you include resource IDs:
```
Just avoiding repitiion since "hosted" is part of the ECH product name.

    
    * {{serverless-full}} Project ID or {{kib}} URL
    * {{ech}} Deployment ID or {{kib}} URL

>   * If hosting on {{ece}} or {{eck}}:

Reviewer 2: ```suggestion
  * If you're hosting on {{ece}} or {{eck}}:
```

    * {{ece}} [diagnostic](/troubleshoot/deployments/cloud-enterprise/run-ece-diagnostics-tool.md) flagging `--deployments` as applicable
    * {{eck}} [diagnostic](/troubleshoot/deployments/cloud-on-k8s/run-eck-diagnostics.md)

  * If self-hosting:

>     * {{es}} [diagnostic](troubleshoot/elasticsearch/diagnostic.md) and [debug logs](deploy-manage/monitor/logging-configuration/update-elasticsearch-logging-levels.md)

Reviewer 2: ```suggestion
    * {{es}} [diagnostic](/troubleshoot/elasticsearch/diagnostic.md) and [debug logs](/deploy-manage/monitor/logging-configuration/update-elasticsearch-logging-levels.md)
```
I think this should fix some of those build errors.

>     * {{kib}} and {{fleet}} [diagnostic](troubleshoot/kibana/capturing-diagnostics.md) and [debug logs](/docs/deploy-manage/monitor/logging-configuration/kibana-log-levels.md)

Author: ```suggestion
    * {{kib}} and {{fleet}} [diagnostic](/troubleshoot/kibana/capturing-diagnostics.md) and [debug logs](/docs/deploy-manage/monitor/logging-configuration/kibana-log-levels.md)
```

    
    * Ingest tools

>       * {{ls}} [diagnostic](troubleshoot/ingest/logstash/diagnostic.md) and [debug logs](reference/logstash/logging.md)

Reviewer 2: ```suggestion
      * {{ls}} [diagnostic](/troubleshoot/ingest/logstash/diagnostic.md) and [debug logs](docs-content://logging.md)
```
This should fix the first one. I'm not sure about the logstash one but let's see if this works. :-)

      * {{agent}} [diagnostic](https://www.elastic.co/docs/troubleshoot/ingest/fleet/common-problems#trb-collect-agent-diagnostics) and [debug logs](https://www.elastic.co/docs/reference/fleet/monitor-elastic-agent#change-logging-level)
  

  :::{tip}

>   The {{es}} cluster maintains an advanced task management system, while other products use a simpler polling mechanism. Because of this, some issues appear only in their start-up debug logs and later logs may only note that the subprocess has stopped or that it has not changed state from an earlier error.

Reviewer 2: ```suggestion
  The {{es}} cluster maintains an advanced task management system, while other products use a simpler polling mechanism. Because of this, some issues appear only in their start-up debug logs. Later logs may note only that the subprocess has stopped or that it has not changed state from an earlier error.
```
Breaking this up to avoid having more than one operator in the same clause.

  :::

  :::{warning}
  Diagnostics and logs mainly emit product metadata and settings but may expose sensitive data which needs privacy-redacted before sharing outside of your organization. See each product's diagnostic page for information on sanitizing output.

>   :::

Author: @anniegale9538 thanks for the value-add feedback of sanitization disclaimer! LMK if you think to say any different than just prior (and I will work separately to add how-to sanitize to each product diag page where it may be missing). 🙏

>   Diagnostics and logs mainly emit product metadata and settings but may expose sensitive data which needs to be redacted before sharing outside of your organization. See each product's diagnostic page for information on sanitizing output.

Reviewer 2: ```suggestion
  Diagnostics and logs mainly emit product metadata and settings, but they may expose sensitive data which needs to be redacted before being shared outside of your organization. Refer to each product's diagnostics page for information on sanitizing output.
```
